tensorflow/compiler/xla/pjrt/pjrt_plugin_device_client.h

```diff
@@ -0,0 +1,32 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_COMPILER_XLA_PJRT_PLUGIN_DEVICE_CLIENT_H_
+#define TENSORFLOW_COMPILER_XLA_PJRT_PLUGIN_DEVICE_CLIENT_H_
+
+#include <memory>
+
+#include "tensorflow/compiler/xla/statusor.h"
+
+namespace xla {
+
+class PjRtClient;
+
+// Not implemented by default. It is the responsibility of the plugin device
+// author to provide an implementation of this function.
```

Comment:
How/where is the plugin linked in? Is the idea you would build your own PJRT client binary that would also include the plugin client definition?

Reply:
Yes, correct. It would be linked through `//tensorflow/compiler/plugin:plugin`, which is reserved for third-party XLA backends. This is exactly where the IPU XLA backend lives in our fork. Here's an example of a minimal dummy implementation to make it compile and link https://github.com/jakeh-gc/tensorflow/commit/c9772a1017d8b83aeb5e72044578336e31c49def

Reply:
> It would be linked through //tensorflow/compiler/plugin:plugin, which is reserved for third-party XLA backends.

Can you mention this in the comment as well?